### ISSUE TITLE
Array UUID not parsed by client ExecuteNonQueryAsync by ParameterizedArray(UUID)

### ISSUE THREAD

**Issue by NpDen4ik** [DarkWanderer/ClickHouse.Client#115]
Hello sorry for my bad language.

Example:
```csharp
[Test]
public async Task ShouldInsertParameterizedArray()
{
    await connection.ExecuteStatementAsync("TRUNCATE TABLE IF EXISTS test.uuid_array");
    await connection.ExecuteStatementAsync(
        "CREATE TABLE IF NOT EXISTS test.uuid_array (arr Array(UUID)) ENGINE TinyLog");

    var command = connection.CreateCommand();
    command.AddParameter("values", new[] {Guid.NewGuid(), Guid.NewGuid(),});
    command.CommandText = "INSERT INTO test.uuid_array VALUES ({values:Array(UUID)})";
    await command.ExecuteNonQueryAsync();

    var count = await connection.ExecuteScalarAsync("SELECT COUNT(*) FROM test.uuid_array");
    Assert.AreEqual(1, count);
}
```

Exception:
```
ClickHouse.Client.ClickHouseServerException (0x0000001B): Code: 27. DB::ParsingException: Cannot parse input: expected '\'' before: 'toUUID(\'8a22df62-aaae-4aff-b552-1a28168cd9a2\'),toUUID(\'e1f8dd19-ca37-47b8-8db1-ae66f3d6a246\')]':  at row 0: While executing ValuesBlockInputFormat. (CANNOT_PARSE_INPUT_ASSERTION_FAILED) (version 22.1.2.2 (official build))

   at ClickHouse.Client.ADO.ClickHouseConnection.HandleError(HttpResponseMessage response, String query) in C:\Projects\ClickHouse.Client\ClickHouse.Client\ADO\ClickHouseConnection.cs:line 252
   at ClickHouse.Client.ADO.ClickHouseConnection.PostSqlQueryAsync(String sqlQuery, CancellationToken token, ClickHouseParameterCollection parameters) in C:\Projects\ClickHouse.Client\ClickHouse.Client\ADO\ClickHouseConnection.cs:line 242
   at ClickHouse.Client.ADO.ClickHouseCommand.ExecuteNonQueryAsync(CancellationToken cancellationToken) in C:\Projects\ClickHouse.Client\ClickHouse.Client\ADO\ClickHouseCommand.cs:line 64
   at ClickHouse.Client.Tests.ParameterizedInsertTests.ShouldInsertParameterizedArray()
```

I don't know may be this problem on clickhouse http interface because if you try do this:
![изображение](https://user-images.githubusercontent.com/32520717/150434194-e0e93ad5-6bff-4065-a3c3-27348c82583b.png)

It's work !!!

And you can delete toUUID('uuid')
![изображение](https://user-images.githubusercontent.com/32520717/150434425-e85d093d-591e-40d0-8139-3e14dd388de6.png)

And this worked

---

In my current solution, i just replace Guid[] to String[]:

```csharp
if (parameter.Value is Guid[] guidsValue)
    parameter.Value = guidsValue.Select(e => e.ToString()).ToArray();

// and
 command.CommandText = "INSERT INTO test.uuid_array VALUES ({values:Array(String)})"
```

It resolve problem, but it's bad solution.

Please change logic on Guid[] in:
- HttpParameterFormatter
- InlineParameterFormatter

for format result will be equals: 
```
[toUUID('4953ecd0-d212-4828-8cc5-11dd386f78ca')] => ['4953ecd0-d212-4828-8cc5-11dd386f78ca']
```

**Comment by DarkWanderer** [comment]
Hi,

What version of ClickHouse and ClickHouse.Client you are using, please?

**Comment by NpDen4ik** [comment]
> Hi,
> 
> I see server version is 22.1.2.2. What version of ClickHouse.Client you are using, please?

Hi 
```sql select version(); ```
Result:
```select version();```

I'm using https://hub.docker.com/r/yandex/clickhouse-server (latest)
![изображение](https://user-images.githubusercontent.com/32520717/150515947-733302cc-40d4-407c-9252-cb2151bd5e3b.png)

**Comment by DarkWanderer** [comment]
Thanks. Yes, I see where the issue is - need to tweak nested values formatting, thanks for reporting

**Comment by NpDen4ik** [comment]
> Thanks. Yes, I see where the issue is - need to tweak nested values formatting, thanks for reporting

Ok thank you very much for your work

**Comment by DarkWanderer** [comment]
Released in [4.2.0](https://github.com/DarkWanderer/ClickHouse.Client/releases/tag/4.2.0)
